### ISSUE TITLE
NPE when compiling with SVM after upgrading to RC13

### ISSUE THREAD

**Issue by fniephaus** [oracle/graal#1046]
After upgrading GraalSqueak from RC12 to RC13, I am getting the following `NullPointerException` when compiling it with SubstrateVM:

```
[graalsqueak-svm:10060]    classlist:   8,038.37 ms
[graalsqueak-svm:10060]        (cap):   1,886.64 ms
[graalsqueak-svm:10060]        setup:   4,096.95 ms
[graalsqueak-svm:10060]     analysis: 108,605.38 ms
Fatal error: java.lang.NullPointerException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:598)
	at java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1005)
	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:434)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:284)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:406)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:106)
Caused by: java.lang.NullPointerException
	at org.graalvm.compiler.bytecode.BytecodeStream.setBCI(BytecodeStream.java:209)
	at org.graalvm.compiler.bytecode.BytecodeStream.<init>(BytecodeStream.java:47)
	at org.graalvm.compiler.java.BytecodeParser.<init>(BytecodeParser.java:721)
	at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.<init>(SharedGraphBuilderPhase.java:79)
	at com.oracle.svm.hosted.phases.SubstrateGraphBuilderPhase$SubstrateBytecodeParser.<init>(SubstrateGraphBuilderPhase.java:87)
	at com.oracle.svm.graal.hosted.GraalFeature$RuntimeBytecodeParser.<init>(GraalFeature.java:261)
	at com.oracle.svm.graal.hosted.GraalFeature$RuntimeGraphBuilderPhase.createBytecodeParser(GraalFeature.java:253)
	at org.graalvm.compiler.java.GraphBuilderPhase$Instance.run(GraphBuilderPhase.java:95)
	at org.graalvm.compiler.phases.Phase.run(Phase.java:49)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:197)
	at org.graalvm.compiler.phases.Phase.apply(Phase.java:42)
	at org.graalvm.compiler.phases.Phase.apply(Phase.java:38)
	at com.oracle.svm.graal.hosted.GraalFeature.processMethod(GraalFeature.java:509)
	at com.oracle.svm.graal.hosted.GraalFeature.duringAnalysis(GraalFeature.java:455)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$8(NativeImageGenerator.java:679)
	at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:63)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:679)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:499)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:417)
	at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1386)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Error: Image building with exit status 1
```

It seems that `code` is `null` at:
https://github.com/oracle/graal/blob/9160236d1ae227218b54cbc1bdf5f549d085a1bb/compiler/src/org.graalvm.compiler.bytecode/src/org/graalvm/compiler/bytecode/BytecodeStream.java#L209

Any suggestions how I can debug this further?

**Comment by cstancu** [comment]
@fniephaus we have seen this issue internally and [fixed at the JVMCI level](https://github.com/graalvm/graal-jvmci-8/commit/54e5821da49a379a4dbf75f9efee9623d13c7f4d). You will need to wait for the next release to get the fix. (cc @dougxc).

The context in which we have seen the problem was when Truffle languages were missing a `@TruffleBoundary` and a native method became reachable for runtime compilation.

**Comment by fniephaus** [comment]
Thanks for the info! Happy to wait, but I'm wondering: must native methods always be annotated with `@TruffleBoundary`?

**Comment by fniephaus** [comment]
Fixed in [`jvmci-0.56`](https://github.com/graalvm/openjdk8-jvmci-builder/releases/tag/jvmci-0.56).
